plugins/woocommerce/includes/wc-template-functions.php

```diff
@@ -1622,7 +1622,6 @@ function wc_get_gallery_image_html( $attachment_id, $main_image = false ) {
 		apply_filters(
 			'woocommerce_gallery_image_html_attachment_image_params',
 			array(
-				'title'                   => _wp_specialchars( get_post_field( 'post_title', $attachment_id ), ENT_QUOTES, 'UTF-8', true ),
```

Comment:
To remove the `title` HTML attribute from the product gallery images, the `title` property needs to be removed from the array returned by the `woocommerce_gallery_image_html_attachment_image_params` filter. However, plugins/themes that have hooked into this filter may depend on this property.

What would be the appropriate way to handle this case?

Reply:
I think the only approach we could take here is to move this filter out of the `wp_get_attachment_image` call and instead assign the value of whatever the filter returns to a variable, then use that variable in `wp_get_attachment_image`. This would then allow us to keep the arguments the same in the filter call but remove `title` after the fact.

Something like the following (note it's untested):
```php
$image_params = apply_filters(
	'woocommerce_gallery_image_html_attachment_image_params',
	array(
		'title'                   => _wp_specialchars( get_post_field( 'post_title', $attachment_id ), ENT_QUOTES, 'UTF-8', true ),
		'data-caption'            => _wp_specialchars( get_post_field( 'post_excerpt', $attachment_id ), ENT_QUOTES, 'UTF-8', true ),
		'data-src'                => esc_url( $full_src[0] ),
		'data-large_image'        => esc_url( $full_src[0] ),
		'data-large_image_width'  => esc_attr( $full_src[1] ),
		'data-large_image_height' => esc_attr( $full_src[2] ),
		'class'                   => esc_attr( $main_image ? 'wp-post-image' : '' ),
	),
	$attachment_id,
	$image_size,
	$main_image
);

if ( isset( $image_params['title'] ) {
	unset( $image_params['title'] );
}

$image = wp_get_attachment_image(
	$attachment_id,
	$image_size,
	false,
	$image_params
);
```

Reply:
Great idea, @dkotter. Thanks!
Please re-review the PR.